pkg/skaffold/build/jib/sync.go

```diff
@@ -46,6 +53,131 @@ type JSONSyncEntry struct {
 	Dest string `json:"dest"`
 }
 
+func InitSync(ctx context.Context, workspace string, a *latest.JibArtifact) error {
+	syncMap, err := getSyncMapFunc(ctx, workspace, a)
+	if err != nil {
+		return err
+	}
+	syncLists[getProjectKey(workspace, a)] = *syncMap
+	return nil
+}
+
+// returns toCopy, toDelete, error
+func GetSyncDiff(ctx context.Context, workspace string, a *latest.JibArtifact, e filemon.Events) (map[string][]string, map[string][]string, error) {
+	// if anything that was modified was a buildfile, do NOT sync, do a rebuild
+	buildFiles := GetBuildDefinitions(workspace, a)
+	for _, f := range e.Modified {
+		f, err := toAbs(f)
+		if err != nil {
+			return nil, nil, errors.WithStack(err)
+		}
+		for _, bf := range buildFiles {
+			if f == bf {
+				return nil, nil, nil
+			}
+		}
+	}
+
+	// no deletions
+	if len(e.Deleted) != 0 {
+		// change into logging
+		fmt.Println("Deletions are not supported by jib auto sync at the moment")
+		return nil, nil, nil
+	}
+
+	currSyncMap := syncLists[getProjectKey(workspace, a)]
+
+	// if all files are modified and direct, we don't need to build anything
```

Comment:
i did not understand the comment. did you mean "all files modified are direct"?

Reply:
updated the comment